core/src/main/java/com/linecorp/armeria/common/util/Version.java

```diff
@@ -71,6 +78,36 @@
     private static final String PROP_LONG_COMMIT_HASH = ".longCommitHash";
     private static final String PROP_REPO_STATUS = ".repoStatus";
 
+    private static final Map<ClassLoader, Map<String, Version>> VERSIONS =
+            new MapMaker().weakKeys().makeMap();
+
+    /**
+     * Returns the version information for the Armeria artifact named {@code artifactId}. If information for
+     * the artifact can't be found, a default value is returned with arbitrary {@code unknown} values.
+     */
+    public static Version get(String artifactId) {
+        return get(artifactId, Version.class.getClassLoader());
+    }
+
+    /**
+     * Returns the version information for the Armeria artifact named {@code artifactId} using the specified
+     * {@link ClassLoader}. If information for the artifact can't be found, a default value is returned
+     * with arbitrary {@code unknown} values.
+     */
+    public static Version get(String artifactId, ClassLoader classLoader) {
+        final Version version = identify(classLoader).get(artifactId);
+        if (version != null) {
+            return version;
+        }
+        return new Version(
+                artifactId,
+                "unknown",
+                0,
+                "unknown",
+                "unknown",
+                "unknown");
```

Comment:
nit: make it singleton?

Reply:
Unfortunately, it has `artifactId` mainly for use in DocService. I could special-case docservice itself but since this is an unexpected case, I figured it's fine to keep it simple like this.

Reply:
This is mainly used for DocsService, LGTM 👍